src/lib/components/Swap/SwapButton.tsx

```diff
@@ -64,9 +69,21 @@ export default function SwapButton({ disabled }: SwapButtonProps) {
     return { disabled: true }
   }, [approval, disabled, inputCurrencyAmount, inputCurrencyBalance])
 
+  // @TODO(ianlapham): connect deadline from state instead of passing undefined.
+  const { signatureData } = useERC20PermitFromTrade(optimizedTrade, allowedSlippage, undefined)
+
+  // the callback to execute the swap
+  const { callback: swapCallback } = useSwapCallback(optimizedTrade, allowedSlippage, account ?? null, signatureData)
+
+  //@TODO(ianlapham): add a loading state, process errors
   const onConfirm = useCallback(() => {
-    // TODO(zzmp): Transact the trade.
-  }, [])
+    if (!swapCallback) {
+      return
+    }
+    swapCallback().then((hash) => {
+      console.log(hash)
```

Comment:
Mark this with two TODOs:
- `// TODO(ianlapham): Add the swap tx to transactionsAtom`
- `// TODO(ianlapham): Add the pending swap tx to a new swap state`